tests/storage/hotplug.go

```diff
@@ -156,9 +156,19 @@ var _ = SIGDescribe("Hotplug", func() {
 	}
 
 	addVolumeVMWithSource := func(name, namespace string, volumeOptions *v1.AddVolumeOptions) {
-		Eventually(func() error {
-			return virtClient.VirtualMachine(namespace).AddVolume(context.Background(), name, volumeOptions)
-		}, 3*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		var err error
+
+		// Try at least 3 times, this is done because `AddVolume` is inherently racy in the way it's implemented
+		// as when it patches the VM Status it expects the field `volumeRequests` to be there (by using a test json patch op),
+		// but at the same time virt-controller trims this field when all requests have been satified.
+		// To avoid hitting these we should explicitly try multiple times.
+		for i := 0; i < 3; i++ {
+			err = virtClient.VirtualMachine(namespace).AddVolume(context.Background(), name, volumeOptions)
+			if err == nil {
+				break
```

Comment:
nit: `break` -> `return`

Reply:
if you do `return` you'd skip the `Expect`.